tests/test_utils_types.py

```diff
@@ -133,3 +136,23 @@ def test_get_metavar(self):
         which is the resource name, but in uppercase.
         """
         self.assertEqual(self.related.get_metavar(None), 'USER')
+
+
+class GeneralTests(unittest.TestCase):
+    """A set of tests that are not specific to any type.
+    """
+    def test_type_name(self):
+        """Establish that custom types are equipped with __name__ field and the
+        upstream click.Choice's issue of having no __name__ field is properly
+        dealt with.
+        """
+        from os import walk
+        from tower_cli import get_resource
```

Comment:
We would like to keep import statements at the top of the file. Also, below, that's a very inventive solution for grabbing all of the different resources, but I would rather see more consistency with the rest of the code base in order to loop through all types of resources.